js/style/style_layer.js

```diff
@@ -178,20 +178,15 @@ function premultiplyLayer(layer, type) {
         outlineProp = type + '-outline-color',
         color = layer[colorProp],
         haloColor = layer[haloProp],
-        outlineColor = layer[outlineProp],
-        opacity = layer[type + '-opacity'];
+        outlineColor = layer[outlineProp];
 
-    var colorOpacity = color && (opacity * color[3]);
-    var haloOpacity = haloColor && (opacity * haloColor[3]);
-    var outlineOpacity = outlineColor && (opacity * outlineColor[3]);
-
-    if (colorOpacity !== undefined && colorOpacity < 1) {
-        layer[colorProp] = util.premultiply([color[0], color[1], color[2], colorOpacity]);
+    if (color) {
+        layer[colorProp] = util.premultiply([color[0], color[1], color[2], color[3] * 255]);
```

Comment:
hmm, maybe this isn't the clearest place to do the multiplication by 255. The result is correct because multiplying the alpha value by 255 before premultiplication is the same as multiplying all components by 255 after premultiplication. I can move it elsewhere tomorrow.